src/editor/build/build_lock_watcher.cpp

```diff
@@ -0,0 +1,66 @@
+#ifdef TOOLS_ENABLED
+
+#include <core/os/dir_access.h>
+#include <modules/kotlin_jvm/src/gd_kotlin.h>
+#include <modules/kotlin_jvm/src/path_provider.h>
+#include <editor/editor_settings.h>
+#include "build_lock_watcher.h"
+
+void BuildLockWatcher::poll_build_lock(void* p_userdata) {
+    jni::Jvm::attach();
+    while (BuildLockWatcher::get_instance().polling_thread_running) {
+        if (!FileAccess::exists(PathProvider::provide_build_lock_file_path())) {
+            reload_if_needed();
+        }
+        Variant poll_delay_variant = _EDITOR_GET("kotlin_jvm/editor/build_lock_watch_interval");
+        float poll_delay = 1;
+        if (poll_delay_variant.get_type() == Variant::REAL || poll_delay_variant.get_type() == Variant::INT) {
+            poll_delay = poll_delay_variant;
+        }
+        OS::get_singleton()->delay_usec(static_cast<int>(poll_delay * 1000 * 1000));
+    }
+    jni::Jvm::detach();
+}
+
+void BuildLockWatcher::reload_if_needed() {
+    String build_usercode_path = PathProvider::provide_build_usercode_path();
```

Comment:
I would use const ref here for const correctness and give indication to future developper.